lib/src/widgets/atomic/report_dialogue.dart

```diff
@@ -0,0 +1,131 @@
+  /// The ID of the item being reported
+  final String itemID;
+
+  /// Constructs a new report dialogue widget
+  const ReportDialogue({required this.itemType, required this.itemID, super.key});
+
+  @override
+  State<ReportDialogue> createState() => _ReportDialogueState();
+}
+
+/// A bunch of reasons that will appear on the dropdown menu,
+final reportReasons = [
+  "Other",
+  "Inappropriate/NSFW",
+];
+
+/// These ones depends on the type of item being reported
+final specificReportReasons = [
+  (
+    reason: "Troll", 
+    applicableTypes: [
+      ItemType.product, 
+      ItemType.sellerProfile, 
+      ItemType.userProfile,
+    ]
+  ),
+  (
+    reason: "Fraudulent activity", 
+    applicableTypes: [
+      ItemType.userProfile, 
+      ItemType.sellerProfile,
+    ]
+  ),
+  (
+    reason: "Disrespectful", 
+    applicableTypes: [
+      ItemType.userProfile, 
+      ItemType.sellerProfile,
+      ItemType.conversation,
+    ]
+  ),
+];
+
+class _ReportDialogueState extends State<ReportDialogue> {
+  final TextEditingController _textController = TextEditingController();
+  late String selectedReason;
+  late final List<String> choices;
+
+  @override
+  void initState() {
+    super.initState();
+    
+    choices = [
+      ...reportReasons, 
+      ...specificReportReasons
+        .where((record) => record.applicableTypes.contains(widget.itemType))
+        .map((record) => record.reason),
+    ];
+    
+    selectedReason = choices.first;
+  }
+
+  Future<void> submit() {
+    final report = Report(
+      reason: selectedReason,
+      comment: _textController.text,
+      author: services.auth.userID!,
+      itemID: widget.itemID,
+      type: widget.itemType,
+      id: services.database.reports.newID,
+    );
+    
+    return services.database.saveReport(report);
+  }
+
+  @override
+  Widget build(BuildContext context) => AlertDialog(
+    title: Text("Report this ${widget.itemType}"),
+    content: SingleChildScrollView(
+      child: ListBody(
+        children: <Widget>[
+          DropdownButton<String>(
+            value: selectedReason,
+            items: choices.map(
+              (reason) => DropdownMenuItem<String>(
+                value: reason,
+                child: Text(reason),
+              ),
+            ).toList(),
+            onChanged: (newVal) { 
+              setState(() {
+                selectedReason = newVal!; 
+              });
+            },
+          ),
+          TextField(
+            maxLines: 2,
+            controller: _textController,
+            decoration: const InputDecoration(
+              labelText: "Comments",
+            ),
+          ),
+        ],
+      ),
+    ),
+    actions: <Widget>[
+      TextButton(
+        child: const Text("Submit"),
+        onPressed: () {
+          submit().whenComplete(
+            () {
+              if (context.mounted) {
+                Navigator.of(context).pop();
+              }
+            }
+          );
+        },
```

Comment:
Behold, async/await: 

```suggestion
        onPressed: () async {
          await submit();
          if (!context.mounted) return;
          Navigator.of(context).pop();
        }
```

Note that this is one of those things you'll need to do even with a view model, since you're using the context and fundamentally updating the UI with no logic. So the view model approach would be: 
```dart
        onPressed: () async {
          await model.submit();
          if (!context.mounted) return;
          Navigator.of(context).pop();
        }
```